src/coreComponents/managers/TimeHistory/PackCollection.cpp

```diff
@@ -32,29 +37,27 @@ void PackCollection::InitializePostSubGroups( Group * const group )
 
 HistoryMetadata PackCollection::getMetadata( ProblemManager & pm, localIndex collectionIdx )
 {
-  DomainPartition const & domain = *(pm.getDomainPartition( ));
-  MeshLevel const & meshLevel = *domain.getMeshBody( 0 )->getMeshLevel( 0 );
-  Group const * target_object = meshLevel.GetGroupByPath( m_objectPath );
+  DomainPartition & domain = *(pm.getDomainPartition( ));
+  Group const * target_object = this->getTargetObject( domain );
   WrapperBase const * target = target_object->getWrapperBase( m_fieldName );
   if( m_setNames.size() != 0 )
   {
     GEOSX_ERROR_IF( collectionIdx >= m_setNames.size(), "Invalid collection index specified." );
-    localIndex num_indices = m_setsIndices[ collectionIdx ].size( );
+    localIndex num_indices = m_setSizeOverride > 0 ? m_setSizeOverride : m_setsIndices[ collectionIdx ].size( );
```

Comment:
```suggestion
    localIndex num_indices = m_setSizeOverride > m_setsIndices[ collectionIdx ].size( ) ? m_setSizeOverride : m_setsIndices[ collectionIdx ].size( );
```
Don't you want the larger of the two values?